docs/userguide/storagedriver/btrfs-driver.md

```diff
@@ -0,0 +1,266 @@
+
+2. Subsequent image layers are stored as a Btrfs snapshot of the parent layer's subvolume or snapshot.
+
+    The diagram below shows a three-layer image. The base layer is a subvolume. Layer 1 is a snapshot of the base layer's subvolume. Layer 2 is a snapshot of Layer 1's snapshot.
+
+    ![](images/btfs_constructs.jpg)
+
+## Image and container on-disk constructs
+
+Image layers and containers are visible in the Docker host's filesystem at `/var/lib/docker/btrfs/subvolumes/<image-id> OR <container-id>`. Directories for containers are present even for containers with a stopped status. This is because the `btrfs` storage driver mounts a default, tob-level subvolume at `/var/lib/docker/subvolumes`. All other subvolumes and snapshots exist below that as Btrfs filesystem objects and not as individual mounts.
+
+The following example shows a single Docker image with four image layers.
+
+```bash
+$ sudo docker images -a
+REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
+ubuntu              latest              0a17decee413        2 weeks ago         188.3 MB
+<none>              <none>              3c9a9d7cc6a2        2 weeks ago         188.3 MB
+<none>              <none>              eeb7cb91b09d        2 weeks ago         188.3 MB
+<none>              <none>              f9a9f253f610        2 weeks ago         188.1 MB
+```
+
+Each image layer exists as a Btrfs subvolume or snapshot with the same name as it's image ID as illustrated by the `btrfs subvolume list` command shown below:
+
+```bash
+$ sudo btrfs subvolume list /var/lib/docker
+ID 257 gen 9 top level 5 path btrfs/subvolumes/f9a9f253f6105141e0f8e091a6bcdb19e3f27af949842db93acba9048ed2410b
+ID 258 gen 10 top level 5 path btrfs/subvolumes/eeb7cb91b09d5de9edb2798301aeedf50848eacc2123e98538f9d014f80f243c
+ID 260 gen 11 top level 5 path btrfs/subvolumes/3c9a9d7cc6a235eb2de58ca9ef3551c67ae42a991933ba4958d207b29142902b
+ID 261 gen 12 top level 5 path btrfs/subvolumes/0a17decee4139b0de68478f149cc16346f5e711c5ae3bb969895f22dd6723751
+```
+
+Under the `/var/lib/docker/btrfs/subvolumes` directoy, each of these subvolumes and snapshots are visible as a normal Unix directory:
+
+```bash
+$ ls -l /var/lib/docker/btrfs/subvolumes/
+total 0
+drwxr-xr-x 1 root root 132 Oct 16 14:44 0a17decee4139b0de68478f149cc16346f5e711c5ae3bb969895f22dd6723751
+drwxr-xr-x 1 root root 132 Oct 16 14:44 3c9a9d7cc6a235eb2de58ca9ef3551c67ae42a991933ba4958d207b29142902b
+drwxr-xr-x 1 root root 132 Oct 16 14:44 eeb7cb91b09d5de9edb2798301aeedf50848eacc2123e98538f9d014f80f243c
+drwxr-xr-x 1 root root 132 Oct 16 14:44 f9a9f253f6105141e0f8e091a6bcdb19e3f27af949842db93acba9048ed2410b
+```
+
+Because Btrfs works at the filesystem level and not the block level, each image and container layer you can browse them in the filesystem using normal Unix commands. The example below shows a truncated output of an `ls -l` command against the image's top layer:
```

Comment:
"Because Btrfs works at the filesystem level and not the block level, each image and container layer can be browsed in the filesystem using normal Unix commands."